eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -1,123 +1,136 @@
 # Add 'AzsdkResourceType' member to outputs since actual output types have changed over the years.
 
 function Get-PurgeableGroupResources {
-    param (
-        [Parameter(Mandatory=$true, Position=0)]
-        [string] $ResourceGroupName
-    )
-
-    # Get any Key Vaults that will be deleted so they can be purged later if soft delete is enabled.
-    Get-AzKeyVault @PSBoundParameters | ForEach-Object {
-        # Enumerating vaults from a resource group does not return all properties we required.
-        Get-AzKeyVault -VaultName $_.VaultName | Where-Object { $_.EnableSoftDelete } `
-            | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Key Vault' -PassThru
-    }
-
-    # Get any Managed HSMs in the resource group, for which soft delete cannot be disabled.
-    Get-AzKeyVaultManagedHsm @PSBoundParameters `
-        | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Managed HSM' -PassThru
+  param (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string] $ResourceGroupName
+  )
+  $purgeableResources = @()
+
+  Write-Verbose "Retrieving deleted Key Vaults from resource group $ResourceGroupName"
+
+  # Get any Key Vaults that will be deleted so they can be purged later if soft delete is enabled.
+  $deletedKeyVaults = Get-AzKeyVault -ResourceGroupName $ResourceGroupName -ErrorAction Ignore | ForEach-Object {
+    # Enumerating vaults from a resource group does not return all properties we required.
+    Get-AzKeyVault -VaultName $_.VaultName | Where-Object { $_.EnableSoftDelete } `
```

Comment:
I wonder if the `-ErrorAction Ignore` should be here as well in the off chance - but possible - even it was deleted between these REST calls. I checked and it still terminates the pipe on this line (i.e. so `where-object` doesn't complain about an empty pipe) so it would be safe.

Reply:
I will add it here as well.